pkg/registries/adapters/adapter.go

```diff
@@ -130,3 +138,50 @@ func imageToSpec(log *logging.Logger, req *http.Request, apbtag string) (*apb.Sp
 
 	return spec, nil
 }
+
+func isCompatibleVersion(specVersion string, minVersion string, maxVersion string) bool {
+	if len(strings.Split(specVersion, ".")) != 2 || len(strings.Split(minVersion, ".")) != 2 || len(strings.Split(maxVersion, ".")) != 2 {
+		return false
+	}
+	specMajorVersion, err := strconv.Atoi(strings.Split(specVersion, ".")[0])
+	if err != nil {
+		return false
+	}
+	specMinorVersion, err := strconv.Atoi(strings.Split(specVersion, ".")[1])
+	if err != nil {
+		return false
+	}
+	minMajorVersion, err := strconv.Atoi(strings.Split(minVersion, ".")[0])
+	if err != nil {
+		return false
+	}
+	minMinorVersion, err := strconv.Atoi(strings.Split(minVersion, ".")[1])
```

Comment:
How come we have to check minor versions? I thought we accept APBVersions >= 1.0 and < 2.0.

Reply:
We need to check minor versions because of the scenario where we support beyond 2.0. We need a way to check that the version fits outside of one major version bump.

Reply:
So you're saying we'd support >= 1.0 and < 2.x? If we do that then we can't really say that _only_ a major version bump will break the API.  IMO, that goes against the reason to have major vs minor versioning.

Reply:
I'm not sure how we can't say that? A broker that only knows how to deploy 1.X APBs will never be able to support 2.X APBs. However, I would expect a broker that knows how to deploy 2.X APBs to know how to deploy 1.X APBs as that would be supporting backwards compatibility. In this case the major version bump *does* break the API, it's just that we expect the broker to support 1.X APBs even as we move to a 2.X spec.

Reply:
> In this case the major version bump does break the API, it's just that we expect the 
> broker to support 1.X APBs even as we move to a 2.X spec.
>
👍 

Here's an example of what I'm thinking:
- We move to APBVersion 2.3.
- APBVersion >= 1.0 and APBVersion < 3.0 are supported
   
Why would we need to check the minor version when we know that anything greater than or equal to the minimal major version and less than the maximum major version is valid? In other words, all breaking changes are per major version. If we support 1.0 then we also support 1.7 and 1.x so there's no reason to check beyond the major version.

Reply:
I see. Using your example, let's take a broker that is installed with 2.3 APBs. We then release a new version 2.4 (and while the upgrade to 2.4 will *not* be a breaking change, it would possibly expose a new option or spelling change). 

I see a use case where an installed broker will not support the new 2.4 option and may not want to deploy APBs that expose that option. It is still possible for the broker to be configured to have 1.0-3.0 as an acceptable range but gives the administrator the ability to fine tune what versions they want to accept and list. I don't see this being an immediate problem but I see this as something we will have to deal with as different versions of the service-catalog expose different metadata options. These are examples of non breaking changes but something the administrator may want to control.

It seems strange to me that we wouldn't be doing any validation on the minor version because we could expose new options that are *not* API breaking but might not be used in a 2.2 broker for example.

Reply:
Unless you feel that my example would imply a major version bump... then I could see why it would be trivial to check the minor version. I just worry about not checking the minor version at all and someone putting 1.Z which doesn't exist and has no associated schema.

Reply:
> I just worry about not checking the minor version at all and someone putting 1.Z which doesn't 
> exist and has no associated schema.
>
I understand your concern, but if we follow the release plan correctly then we won't have this issue. And if we make a mistake then we can always change the version.

Reply:
I wouldn't mind having Ryan's logic be the default, but if a user has defined some config then we override? This might just be a feature we add later and for now, just follow Ryan's logic for now.

Reply:
+1 sounds good to me

Reply:
+1 @rthallisey example is how I mentally pictured how this would work. The scenario of 2.4 causing a problem means IMO we failed with the change APB spec change.